app/src/main/java/unipd/se18/ocrcamera/IngredientDetailsFragment.java

```diff
@@ -0,0 +1,171 @@
+
+            // Make request to wikipedia, searching for ingredient informations.
+            assert inciName != null;
+            final String url = "https://en.wikipedia.org/w/api.php?format=json&action=query&prop=extracts&" +
+                    "exintro&explaintext&redirects=1&titles="+inciName.toLowerCase();
+            JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
+                    (Request.Method.GET, url, null, new Response.Listener<JSONObject>() {
+
+                        @Override
+                        public void onResponse(JSONObject response) {
+
+                            try {
+                                JSONObject query = (JSONObject) response.get("query");
+                                JSONObject pages = (JSONObject) query.get("pages");
+                                Iterator<String> keys = pages.keys();
+
+                                while(keys.hasNext()) {
+                                    String key = keys.next();
+                                    if (!key.equals("-1") && pages.get(key) instanceof JSONObject) {
+                                        //show wikipedia extract
+                                        JSONObject page = (JSONObject) pages.get(key);
+                                        String wikipediaExtract = (String) page.get("extract");
+                                        wikipediaView.setText(wikipediaExtract);
+                                    } else {
+                                        //wikipedia page not found
+                                        wikipediaView.setText(R.string.wikipedia_not_found);
+                                    }
+                                }
+
+                            } catch (JSONException e) {
+                                Log.e(TAG, "invalid response from wikipedia");
+                                wikipediaView.setText(R.string.wikipedia_failed_request);
+                            }
+
+                        }
+                    }, new Response.ErrorListener() {
+
+                        @Override
+                        public void onErrorResponse(VolleyError error) {
+                            Log.e(TAG, "Could not send request to wikipedia");
+                            wikipediaView.setText(R.string.wikipedia_failed_request);
+                        }
+                    });
+
+            // Add the request to the RequestQueue.
+            RequestQueueSingleton.getInstance(getActivity()).addToRequestQueue(jsonObjectRequest);
+        }
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        //set dialog fragment size (width and height values in fragment_ingredients_details.xml do not work)
+        Window window = getDialog().getWindow();
+        if (window != null) {
+            window.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+        }
+    }
+}
```

Comment:
Hardcoding at line 115, 142,151

Reply:
For the line number 115 you are right, I will create a method `String buildUrlRequest(String inciname)`, thank you. 
As for number 142 and 151 I think it i little bit excessive and unnecessary to create a constant for every log message for this reason: these are just log messages for the programmer and it is not visualized in any way by the user, moreover these strings are used just once in the code. It would be easier to put the message directly in the log, this way we can avoid going up and down reading the code. What do you think?